helm/provider.go

```diff
@@ -233,9 +233,8 @@ func kubernetesResource() *schema.Resource {
 func providerConfigure(d *schema.ResourceData, terraformVersion string) (interface{}, error) {
 	m := &Meta{data: d}
 
-	settings := &cli.EnvSettings{
-		Debug: d.Get("debug").(bool),
-	}
+	settings := cli.New()
+	settings.Debug = d.Get("debug").(bool)
```

Comment:
Is there any case in which "debug" might not exist? This change doesn't protect against that case, as far as I was able to see. I tested it by replacing "debug" with something I know doesn't exist:

```
[dakini@dax terraform-provider-helm (fix-plugin-crash *%=)]$ git diff
diff --git a/helm/provider.go b/helm/provider.go
index 0ccfcf955..12e6e4dc8 100644
--- a/helm/provider.go
+++ b/helm/provider.go
@@ -234,7 +234,7 @@ func providerConfigure(d *schema.ResourceData, terraformVersion string) (interfa
        m := &Meta{data: d}

        settings := cli.New()
-       settings.Debug = d.Get("debug").(bool)
+       settings.Debug = d.Get("potato").(bool)

        if v, ok := d.GetOk("plugins_path"); ok {
                settings.PluginsDirectory = v.(string)
[dakini@dax terraform-provider-helm (fix-plugin-crash *%=)]$ make testacc
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test "./helm" -v  -timeout 120m -parallel=4
=== RUN   TestAccDataRepository_basic
--- FAIL: TestAccDataRepository_basic (0.00s)
panic: interface conversion: interface {} is nil, not bool [recovered]
        panic: interface conversion: interface {} is nil, not bool

goroutine 25 [running]:
testing.tRunner.func1.1(0x205f1a0, 0xc0008a9590)
        /home/dakini/usr/go/src/testing/testing.go:1057 +0x30d
testing.tRunner.func1(0xc0006c2d80)
        /home/dakini/usr/go/src/testing/testing.go:1060 +0x41a
panic(0x205f1a0, 0xc0008a9590)
        /home/dakini/usr/go/src/runtime/panic.go:969 +0x175
github.com/hashicorp/terraform-provider-helm/helm.providerConfigure(0xc0009ac070, 0x2329fa8, 0xf, 0xc0009ac070, 0x0, 0x0, 0x0)
        /home/dakini/go/src/github.com/hashicorp/terraform-provider-helm/helm/provider.go:237 +0x413
github.com/hashicorp/terraform-provider-helm/helm.Provider.func2(0xc0009ac070, 0x0, 0xc0009e4fc0, 0xc0009ac070, 0x0)
        /home/dakini/go/src/github.com/hashicorp/terraform-provider-helm/helm/provider.go:111 +0x58
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Configure(0xc000518900, 0xc0008a9200, 0x422552be88081201, 0xffffffffffffffff)
        /home/dakini/go/src/github.com/hashicorp/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/provider.go:275 +0x102
github.com/hashicorp/terraform-provider-helm/helm.testAccPreCheck(0xc0006c2d80, 0x0, 0x0)
        /home/dakini/go/src/github.com/hashicorp/terraform-provider-helm/helm/provider_test.go:73 +0x95
github.com/hashicorp/terraform-provider-helm/helm.TestAccDataRepository_basic.func1()
        /home/dakini/go/src/github.com/hashicorp/terraform-provider-helm/helm/data_repository_test.go:12 +0x32
github.com/hashicorp/terraform-plugin-sdk/helper/resource.Test(0x26a8960, 0xc0006c2d80, 0x0, 0xc0008d93e0, 0xc00070f1d0, 0x0, 0x0, 0x0, 0xc000533900, 0x3, ...)
        /home/dakini/go/src/github.com/hashicorp/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/resource/testing.go:597 +0x191f
github.com/hashicorp/terraform-provider-helm/helm.TestAccDataRepository_basic(0xc0006c2d80)
        /home/dakini/go/src/github.com/hashicorp/terraform-provider-helm/helm/data_repository_test.go:11 +0x758
testing.tRunner(0xc0006c2d80, 0x2420c80)
        /home/dakini/usr/go/src/testing/testing.go:1108 +0xef
created by testing.(*T).Run
        /home/dakini/usr/go/src/testing/testing.go:1159 +0x386
FAIL    github.com/hashicorp/terraform-provider-helm/helm       0.045s
FAIL
make: *** [GNUmakefile:30: testacc] Error 1
```

It might be more robust to check that the value exists first, similar to the statements beneath this one, with the `if v, ok ...` syntax.

Reply:
Yeah you're right, this is what it was doing before but it could be more robust. AFAIK if something is defined as TypeBool in the schema it will default to `false` though.

Reply:
ok, I see where this is set now. The helm library makes sure it exists by running it through `strconv.ParseBool`. It throws an error if the user accidentally typos the value:

```
$ HELM_DEBUG=ture make testacc
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test "./helm" -v  -timeout 120m -parallel=4
=== RUN   TestAccDataRepository_basic
--- FAIL: TestAccDataRepository_basic (0.00s)
panic: Error reading level config: strconv.ParseBool: parsing "ture": invalid syntax [recovered]
        panic: Error reading level config: strconv.ParseBool: parsing "ture": invalid syntax

goroutine 12 [running]:
testing.tRunner.func1.1(0x1ff99c0, 0xc00021f9e0)
        /home/dakini/usr/go/src/testing/testing.go:1057 +0x30d
testing.tRunner.func1(0xc0006c3800)
        /home/dakini/usr/go/src/testing/testing.go:1060 +0x41a
panic(0x1ff99c0, 0xc00021f9e0)
        /home/dakini/usr/go/src/runtime/panic.go:969 +0x175
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*ResourceData).get(0xc000928a10, 0xc00021f930, 0x1, 0x1, 0x12, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/dakini/go/src/github.com/hashicorp/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource_data.go:540 +0x399
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*ResourceData).getChange(0xc000928a10, 0x23192da, 0x5, 0xc000921201, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/dakini/go/src/github.com/hashicorp/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource_data.go:511 +0x125
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*ResourceData).diffChange(0xc000928a10, 0x23192da, 0x5, 0x30, 0x30, 0xc000920f50, 0x4ed39a, 0x30)
        /home/dakini/go/src/github.com/hashicorp/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource_data.go:488 +0x97
github.com/hashicorp/terraform-plugin-sdk/helper/schema.schemaMap.diffString(0xc00043ed20, 0x23192da, 0x5, 0xc0008fc8c0, 0xc000921060, 0x26ac9a0, 0xc000928a10, 0xdd2ab00, 0x330b308a7edc2c0f, 0x0)
        /home/dakini/go/src/github.com/hashicorp/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/schema.go:1310 +0x9d
github.com/hashicorp/terraform-plugin-sdk/helper/schema.schemaMap.diff(0xc00043ed20, 0x23192da, 0x5, 0xc0008fc8c0, 0xc000961460, 0x26ac9a0, 0xc000928a10, 0x0, 0xc000921180, 0xc000921198)
        /home/dakini/go/src/github.com/hashicorp/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/schema.go:917 +0x4cc
github.com/hashicorp/terraform-plugin-sdk/helper/schema.schemaMap.Diff(0xc00043ed20, 0x0, 0xc0009a4270, 0x0, 0x0, 0x0, 0x269e901, 0x0, 0xc0009a41e0, 0xc0009a4180)
        /home/dakini/go/src/github.com/hashicorp/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/schema.go:500 +0x219
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Configure(0xc000696c80, 0xc0009a4270, 0xea4f5a6a63a8da01, 0xffffffffffffffff)
        /home/dakini/go/src/github.com/hashicorp/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/provider.go:260 +0x75
github.com/hashicorp/terraform-provider-helm/helm.testAccPreCheck(0xc0006c3800, 0x0, 0x0)
        /home/dakini/go/src/github.com/hashicorp/terraform-provider-helm/helm/provider_test.go:73 +0x95
github.com/hashicorp/terraform-provider-helm/helm.TestAccDataRepository_basic.func1()
        /home/dakini/go/src/github.com/hashicorp/terraform-provider-helm/helm/data_repository_test.go:12 +0x32
github.com/hashicorp/terraform-plugin-sdk/helper/resource.Test(0x26a8960, 0xc0006c3800, 0x0, 0xc00072aa50, 0xc00043f4d0, 0x0, 0x0, 0x0, 0xc0002e1180, 0x3, ...)
        /home/dakini/go/src/github.com/hashicorp/terraform-provider-helm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/resource/testing.go:597 +0x191f
github.com/hashicorp/terraform-provider-helm/helm.TestAccDataRepository_basic(0xc0006c3800)
        /home/dakini/go/src/github.com/hashicorp/terraform-provider-helm/helm/data_repository_test.go:11 +0x758
testing.tRunner(0xc0006c3800, 0x2420c80)
        /home/dakini/usr/go/src/testing/testing.go:1108 +0xef
created by testing.(*T).Run
        /home/dakini/usr/go/src/testing/testing.go:1159 +0x386
FAIL    github.com/hashicorp/terraform-provider-helm/helm       0.037s
FAIL
```

This seems alright. Better than a nil pointer, anyway.